lxd/device/device_utils_unix.go

```diff
@@ -170,18 +184,28 @@ func UnixDeviceCreate(s *state.State, idmapSet *idmap.IdmapSet, devicesPath stri
 		d.Type = "c"
 	}
 
-	// Get the device owner.
-	if m["uid"] != "" {
-		d.UID, err = strconv.Atoi(m["uid"])
+	if shared.IsTrue(m["ownership.inherit"]) {
+		gid, uid, err := unixDeviceOwnership(srcPath)
 		if err != nil {
-			return nil, fmt.Errorf("Invalid uid %s in device %s", m["uid"], srcPath)
+			return nil, fmt.Errorf("Failed to retrieve host ownership of device %s: %w", srcPath, err)
 		}
-	}
 
-	if m["gid"] != "" {
-		d.GID, err = strconv.Atoi(m["gid"])
-		if err != nil {
-			return nil, fmt.Errorf("Invalid gid %s in device %s", m["gid"], srcPath)
+		d.GID = int(gid)
+		d.UID = int(uid)
+	} else {
+		// Get the device owner.
+		if m["uid"] != "" {
+			d.UID, err = strconv.Atoi(m["uid"])
+			if err != nil {
+				return nil, fmt.Errorf("Invalid uid %s in device %s", m["uid"], srcPath)
```

Comment:
whilst we changing this, please can you use %q for quoting values in errors rather than %q